src/js/dataView/dataset/ImageIcon.js

```diff
@@ -45,8 +45,19 @@ class ImageIcon extends React.Component {
     }
 
     getIconSizes() {
-        var width = this.props.iconSize;
-        return {'width': width, 'max-height': width}
+        let longestSide = this.props.iconSize;
+        let width = this.props.image.data.obj.sizeX;
+        let height = this.props.image.data.obj.sizeY;
+        if (width > height) {
+            let ratio = height / width;
+            width = Math.min(longestSide, width);
+            height = width * ratio;
+        } else {
+            let ratio = width / height;
+            height = Math.min(longestSide, height);
+            width = height * ratio;
+        }
+        return {'width': width, 'height': height}
```

Comment:
This is the approach we are using in webclient Datasets. A couple of thoughts...
 - It would be nice if we didn't rely on the ```image.data``` (e.g. revert to old behaviour if it's not there) since this would allow us not to rely on loading this data in the jstree. E.g. if we are using the centre panel to filter search results (https://github.com/ome/omero-parade/pull/7) 
 - In the webclient Plate layout we use a different approach to resizing Well images which doesn't calculate aspect ratio (so doesn't look as good when ```height > width```) but has the advantage that there is no JavaScript needed to iterate through all the images when zooming, so it's much smoother when you have a LOT of images.
See https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webgateway/static/webgateway/js/ome.plateview.js#L244
And css at https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webgateway/static/webgateway/css/ome.plateview.css#L158
In the devtools it looks like:
```
.wellSize64 img {
    width: 64px;
    max-height: 64px;
}
```
I was thinking of using this approach in ```parade``` too, since it handles large numbers of images better. There may be less verbose ways of achieving the same thing in css, but I don't know.

Reply:
I thought this was going to be a quick fix but in reading the rest of the code and looking at what we are doing with the plate view this is a **huge** can of worms. The code duplication and difference of implementation between `Dataset.js`, `DataTable.js`, and `Well.js` along with the re-use of data from the tree is setting off encapsulation warning bells for me everywhere.

I'm going to close this PR for now and we can come back to it later.